populator-machinery/controller_test.go

```diff
@@ -340,14 +381,48 @@ func compareNotifyMap(want []string, got map[string]*stringSet) error {
 }
 
 func compareObjects(want *vpObjects, got *vpObjects) error {
-	if (want == nil && got != nil) || (want != nil && got == nil) {
+	if want != nil && got == nil {
 		return fmt.Errorf("Expected vpObjects is different from actual vpObjects. Expected %+v, actual %+v", want, got)
 	}
-	if want.pvcPrime != nil {
+
+	if want.onlyComparePVCFinalizers {
+		if !reflect.DeepEqual(want.pvc.Finalizers, got.pvc.Finalizers) {
+			return fmt.Errorf("Expected pvc Finalizers is different from actual pvc Finalizers. Expected %+v\n, actual %+v\n",
+				want.pvc.Finalizers, got.pvc.Finalizers)
+		}
+	} else {
+		if !reflect.DeepEqual(want.pvc.Finalizers, got.pvc.Finalizers) {
```

Comment:
Shouldn't this be comparing the pvcs?

Reply:
Thanks for catching this! This is a typo, I mean to compare PVCs